library/common/network/configurator.cc

```diff
@@ -77,11 +79,13 @@ void Configurator::refreshDns(envoy_network_t network) {
   // this avoids triggering a refresh for a non-current network.
   // Note this does NOT completely prevent parallel refreshes from being triggered in multiple
   // flip-flop scenarios.
-  if (network != preferred_network_.load() || !dns_cache_.has_value()) {
+  if (network != preferred_network_.load()) {
     return;
   }
   // TODO(goaway): track event here or are there existing signals we can use?
-  dns_cache_.value()->forceRefreshHosts();
+  if (auto dns_cache = dns_cache_manager_->lookUpCacheByName(BaseDnsCache)) {
+    dns_cache->forceRefreshHosts();
```

Comment:
check for nullptr

Reply:
Sorry, which pointer do you mean?

Reply:
dns_cache. The returned pointer could be null

Reply:
Oh wait I'm being totally dense. You have this in an if 🤦

Reply:
no worries :)